pytato/array.py

```diff
@@ -2025,6 +2027,91 @@ def eye(N: int, M: Optional[int] = None, k: int = 0,  # noqa: N803
                        shape=(N, M), dtype=dtype, bindings={},
                        axes=_get_default_axes(2))
 
+# }}}
+
+
+# {{{ arange
+
+@dataclass
+class _ArangeInfo:
+    start: Optional[int]
+    stop: Optional[int]
+    step: Optional[int]
+    dtype: Optional[np.dtype[Any]]
+
+
+def arange(*args: Any, **kwargs: Any) -> Array:
+    explicit_dtype = False
+
+    # {{{ argument processing
+
+    inf = _ArangeInfo(
+            start=None,
+            stop=None,
+            step=None,
+            dtype=None)
+
+    # Yuck. Thanks, numpy developers. ;)
+    if isinstance(args[-1], np.dtype):
+        inf.dtype = args[-1]
+        args = args[:-1]
+        explicit_dtype = True
+
+    argc = len(args)
+    if argc == 0:
+        raise ValueError("stop argument required")
+    elif argc == 1:
+        inf.stop = args[0]
+    elif argc == 2:
+        inf.start = args[0]
+        inf.stop = args[1]
+    elif argc == 3:
+        inf.start = args[0]
+        inf.stop = args[1]
+        inf.step = args[2]
+    else:
+        raise ValueError("too many arguments")
```

Comment:
```suggestion
        raise TypeError("arange() takes 0 to 4 positional arguments but"
                        f" {argc} were given")
```
NumPy raises a type error.

Reply:
https://github.com/inducer/pytato/compare/88fb1da90f31c62c97a9da4474437ef1d4fcb2d5..01da3534d17727b3150637291a46565f7ecc8dee